src/main/scala/mesosphere/marathon/core/launcher/TaskOp.scala

```diff
@@ -11,14 +11,11 @@ import org.apache.mesos.{ Protos => MesosProtos }
   */
 sealed trait TaskOp {
   /** The ID of the affected task. */
-  def taskId: Task.Id
+  def taskId: Task.Id = newTask.taskId
   /** The MarathonTask state before this operation has been applied. */
   def oldTask: Option[Task]
-  /**
-    * The MarathonTask state after this operation has been applied.
-    * `None` means that the associated task should be expunged.
-    */
-  def maybeNewTask: Option[Task]
+  /** The TaskStateOp that will lead to the new state after this operation has been applied. */
+  def newTask: TaskStateOp
```

Comment:
shouldn't be called "newTask" in my opinion but "taskChange"/"taskStateOp" or something like that.